### PR TITLE
[funind] Port indfun to the new tactic engine.

### PR DESCRIPTION
**Pull request by ejgallego**
It seems to work [famous last words]

### CODE REVIEW

File: plugins/funind/indfun.ml [hunk 1]
```diff
@@ -37,111 +41,102 @@ let choose_dest_or_ind scheme_info args =
...
+            in
+            match elimination_sort_of_goal gl with
+            | InSProp -> finfo.sprop_lemma
+            | InProp -> finfo.prop_lemma
+            | InSet -> finfo.rec_lemma
+            | InType -> finfo.rect_lemma
+          in
+          let princ =  (* then we get the principle *)
+            try
+              let sigma, princ = Evd.fresh_global (pf_env gl) (project gl) (GlobRef.ConstRef (Option.get princ_option)) in
```
Comment (ppedrot) [inline plugins/funind/indfun.ml]: Why not simply match on `princ_option`?

Comment (ejgallego) [reply plugins/funind/indfun.ml]: Good tip, thanks! I think I've made the code a tad cleaner now.

**Comment by ejgallego** [comment]
I have added a commit removing some `Not_found` throwing, that should help in the future to improve control flow [more can be done of course we need to go step by step]

**Comment by gares** [comment]
The move `find` -> `find_opt` does not seem an improvement to me, since clients seem to match on the option and raise `Not_found` in case it is `None`.

**Comment by ejgallego** [comment]
> The move `find` -> `find_opt` does not seem an improvement to me, since clients seem to match on the option and raise `Not_found` in case it is `None`.

@gares I only ported a few files to the new API, indeed if I have time I could add another commit porting the rest, but indeed they would really benefit from a larger refactoring actually abstracting out the pattern they use so that'd may take a while.

With funind we have to go step by step I think.

**Comment by ejgallego** [comment]
What shall we do with this @ppedrot ?

**Comment by ppedrot** [comment]
To be honest, I don't care about funind. It's most likely going to be phased out in favour of Equations, and the current PR doesn't seem to break any dev tracked in the CI. So this can go in.

**Comment by ejgallego** [comment]
> To be honest, I don't care about funind. It's most likely going to be phased out in favour of Equations, and the current PR doesn't seem to break any dev tracked in the CI. So this can go in.

Yeah, it is not like I care a lot about it, however I need to clean up quite a few stuff on the way it handles constant declaration as it is a heavy user of it, so I have no choice than to submit PRs. Any help with them is appreciated.

**Review by ppedrot — APPROVED** [review]
You have my blessings.

**Comment by ejgallego** [comment]
Merci!

**Pull request merged**